relay-server/src/services/processor.rs

```diff
@@ -167,9 +167,29 @@ macro_rules! processing_group {
 /// Should be used only with groups which are responsible for processing envelopes with events.
 pub trait EventProcessing {}
 
+/// A trait for processing groups that can be dynamically sampled.
+pub trait Sampling {
+    /// Whether dynamic sampling should run under the given project's conditions.
+    fn supports_sampling(project_state: &ProjectState) -> bool;
+
+    fn supports_reservoir_sampling() -> bool;
```

Comment:
Doc comment, could also be an associated const, not sure if that makes sense `Sampling::SUPPORTS_RESERVOIR_SAMPLING`.

Reply:
I kinda like that `supports_sampling` and `supports_reservoir_sampling` are both methods, even though one of them is effectively constant.